rust-runtime/aws-smithy-runtime-api/src/client/runtime_plugin/anonymous_auth.rs

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//! The [AnonymousAuthRuntimePlugin] and supporting code.
+
+use super::RuntimePlugin;
+use crate::client::auth::option_resolver::{
+    StaticAuthOptionResolver, StaticAuthOptionResolverParams,
+};
+use crate::client::auth::{AuthSchemeId, HttpAuthScheme, HttpAuthSchemes, HttpRequestSigner};
+use crate::client::identity::anonymous::AnonymousIdentityResolver;
+use crate::client::identity::{Identity, IdentityResolver, IdentityResolvers};
+use crate::client::interceptors::Interceptors;
+use crate::client::orchestrator::{ConfigBagAccessors, HttpRequest};
+use crate::client::runtime_plugin::BoxError;
+use crate::config_bag::ConfigBag;
+
+const ANONYMOUS_AUTH_SCHEME_ID: AuthSchemeId = AuthSchemeId::new("anonymous");
+
+/// A [RuntimePlugin] to provide anonymous authentication. This runtime plugin sets its own:
+/// - [AuthOptionResolver](crate::client::auth::AuthOptionResolver)
+/// - [AuthOptionResolverParams](crate::client::auth::AuthOptionResolverParams)
+/// - [IdentityResolvers]
+/// - [HttpAuthSchemes]
+///
+/// **The above components will replace any existing ones!** As such, don't use this plugin unless:
+/// - You only need to make anonymous requests, such as when interacting with [Open Data](https://aws.amazon.com/opendata/).
+/// - You're writing orchestrator tests and don't care about authentication.
+pub struct AnonymousAuthRuntimePlugin;
```

Comment:
When I was working on implementing `DefaultEntpointResolver` for the orchestrator, I got [this review comment](https://github.com/awslabs/smithy-rs/pull/2577#discussion_r1169370073) from John. And I'm wondering if the same reasoning is applied here, i.e. whether this struct should be in the `aws-smithy-runtime` crate instead. We can definitely run by John if we want to be sure.

Reply:
Good callout, I'll move it to the runtime crate

Reply:
Good callout, I'll move it to the runtime crate